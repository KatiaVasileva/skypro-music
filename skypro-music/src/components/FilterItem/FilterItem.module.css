.popup {
    position: relative;
}

.popupContainer {
    position: absolute;
    width: 248px;
    max-height: 305px;
    background-color: #313131;
    border-radius: 12px;
    display: flex;
    margin-top: -40px;
    z-index: 1;
}

.popupContainerAuthor {
    margin-left: 100px;
}

.popupContainerYear {
    margin-left: 262px;
    height: 196px;
    width: 221px;
}

.popupContainerGenre {
    margin-left: 425px;
}

.popupBox {
    width: 180px;
    max-height: 237px;
    display: flex;
    margin-top: 20px;
    overflow-y: hidden;
    overflow-y: auto;
    margin: 34px 34px;
    scrollbar-color: #ffffff #4B4949;
    scrollbar-width: thin;
    min-height: 128px;
    min-width: 153px;
}

.popupContent {
    width: 152px;
    max-height: 232px;
    display: flex;
    flex-direction: column;
    gap: 24px;
    min-height: 128px;
}


.popupText {
    font-size: 18px;
    font-weight: 400;
    line-height: 24px;

    &:hover {
        color: #B672FF;
        text-decoration: underline;
        cursor: pointer;
    }
}