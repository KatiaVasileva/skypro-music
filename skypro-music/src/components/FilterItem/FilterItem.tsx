import styles from "./FilterItem.module.css";
import classNames from "classnames";

type FilterItemProps = {
  filterName: string;
  filterContent: Array<string>;
  selectedValues: Array<string>;
  handleFilterItemClick: (filterElement: string) => void;
  handleFilterElementClick?: () => void;
  isFilterElementClicked?: boolean;

};

function FilterItem({
  filterName,
  filterContent,
  selectedValues,
  handleFilterItemClick,
  handleFilterElementClick,
  isFilterElementClicked,
}: FilterItemProps) {

  return (
    <div className={styles.popup}>
      <div
        className={classNames(
          styles.popupContainer,
          filterName === "performer"
            ? styles.popupContainerAuthor
            : filterName === "year"
            ? styles.popupContainerYear
            : styles.popupContainerGenre
        )}
      >
        <div className={styles.popupBox}>
          <div className={styles.popupContent}>
            {filterContent.map((filterElement, index) => (
              <p
                className={classNames(
                  styles.popupText,
                    {
                      [styles.popupTextActive]:
                        selectedValues.includes(filterElement),
                      [styles.popupTextInactive]:
                        filterName === "year" && 
                        !selectedValues.includes(filterElement) && isFilterElementClicked,
                    },
                )}
                key={index}
                onClick={() => {
                  handleFilterItemClick(filterElement);
                  if (handleFilterElementClick) {
                    handleFilterElementClick();
                  }
                  console.log(isFilterElementClicked);
                }}
              >
                {filterElement}
              </p>
            ))}
          </div>
        </div>
      </div>
    </div>
  );
}

export default FilterItem;
