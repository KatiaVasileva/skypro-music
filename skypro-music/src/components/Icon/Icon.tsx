import { IconProps } from "@/types/IconProps.types";

function Icon({ wrapperClass = "", iconClass, name, onClick }: IconProps) {
  return (
    <div className={wrapperClass} onClick={onClick}>
      <svg className={iconClass}> 
        <use xlinkHref={`/img/icon/sprite.svg#${name}`}></use>
      </svg>
    </div>
  );
}

export default Icon;
