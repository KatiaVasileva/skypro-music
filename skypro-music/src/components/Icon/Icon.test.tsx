import ReduxProvider from "@/store/ReduxProvider";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import FilterItem from "../FilterItem/FilterItem";
import Icon from "./Icon";

describe("Icon component", () => {
  it("renders correctly", () => {
    const { container } = render(
      <ReduxProvider>
        <Icon name={"name"} />
      </ReduxProvider>
    );
    expect(container).toMatchSnapshot();
  });
});
