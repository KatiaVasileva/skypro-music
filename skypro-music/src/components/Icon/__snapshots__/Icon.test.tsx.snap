// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon component renders correctly 1`] = `
<div>
  <div
    class=""
  >
    <svg>
      <use
        xlink:href="/img/icon/sprite.svg#name"
      />
    </svg>
  </div>
</div>
`;
